pkg/kvstore/etcd.go

```diff
@@ -243,6 +243,13 @@ func (e *etcdClient) GetLeaseID() client.LeaseID {
 	return l
 }
 
+// closeSession closes the current session.
+func (e *etcdClient) closeSession() {
+	e.RWMutex.RLock()
```

Comment:
I haven't had a chance to really think through this but do we need to block all users of `e.session` until the session is renewed? If so, that's going to be annoyingly mess :/ The only simple way I can think of is to use the renewSession goroutine as a way to give out references to the current session, and this would force-block all other users until it re-establishes it.

Reply:
We can also rely on the operation being retried. We'd have to audit the code whether we have kvstore operations that are not retried on error. I think most of them are.